Server/api/routes/referee/matchups.ts

```diff
@@ -91,6 +91,46 @@ refereeMatchupsRouter.$get<{ matchup: MatchupInterface }>("/:tournamentID/:match
     };
 });
 
+// TODO: Move this to an endpoint in api/routes/matchups.ts later when the matchup info page is implemented for cross-use
+refereeMatchupsRouter.$get<{ messages: MatchupMessageInterface[] }>("/:tournamentID/:matchupID/messages", validateTournament, isLoggedInDiscord, hasRoles([TournamentRoleType.Organizer, TournamentRoleType.Referees]), async (ctx) => {
+    const beforeID = parseInt(parseQueryParam(ctx.query.before) ?? "");
+    const messagesQ = MatchupMessage
+        .createQueryBuilder("message")
+        .innerJoin("message.matchup", "matchup")
+        .innerJoinAndSelect("message.user", "user")
+        .where("matchup.ID = :ID", { ID: ctx.params.matchupID });
+    if (beforeID && !isNaN(beforeID) && beforeID > 0)
+        messagesQ.andWhere("message.ID < :beforeID", { beforeID });
+
+    const messages = await messagesQ
+        .orderBy("message.timestamp", "DESC")
```

Comment:
again, timestamps only have second accuracy, so this could lead to messages (especially banchobot's) in the wrong order. the consistent order here is by ID.

```suggestion
        .orderBy("message.ID", "DESC")
```